src/System.Private.CoreLib/shared/System/Threading/EventWaitHandle.cs

```diff
@@ -39,11 +40,12 @@ public static EventWaitHandle OpenExisting(string name)
                 case OpenExistingResult.PathNotFound:
                     throw new DirectoryNotFoundException(SR.Format(SR.IO_PathNotFound_Path, name));
                 default:
+                    Debug.Assert(result != null, "result should be non-null on success");
```

Comment:
So this actually eliminates the need of using `!` when returning in a non-nullable return method?

Reply:
Yes.  The compiler sees the != null and treats that as evidence that the result isn't null.

Reply:
Yes, makes total sense. Also, makes it clearer when reading the code. Good to know.